### ISSUE TITLE
The highlighting of selected titles broken in right-hand-side navigation

### ISSUE THREAD

**Issue by melindafekete** [commercetools/commercetools-docs-kit#1325]
When using the right-hand-side navigation panel to jump to a section of a given docs page, the highlighting doesn't reflect the clicking action. The heading that is clicked is not highlighted in blue, but instead, the heading above (or no heading at all) is highlighted. 

The left-hand-side navigation panel works as expected. I checked a few pages across the HTTP API site, and some other micro-sited, and this behavior seems consistent. 

https://user-images.githubusercontent.com/14042876/173875603-9a04a84d-2dac-4ff1-8c26-2781f7b8d357.mp4

**Comment by nkuehn** [comment]
Thank you for the precise observation! 

I can reproduce it on the page you are using in the video ( https://docs.commercetools.com/api/projects/products-suggestions in case someone want to test, too). 

By UX design and implementation It does not try to highlight the "click action", it's highlighting the current scroll position in the document.  This, after clicking, should normally be the heading you clicked on but not always because there can be weird content scenarios with multiple headings with nearly no content at the end of the page etc. 

Down to a potential root cause: the index navigation works with an "offset", i.e. it's not moving the clicked heading exactly to the top but a little below because it looks weird when the heading is zero pixels from the top and there can be scenarios where e.g. the top bar would overlay it.    What could be going on here is that the offset is too big and that may lead the "scroll spy" that highlights the current position in the document to think it is still in the section above. 

TL;DR: this needs some more manual tuning and tweaking of the spacings and offsets until we reach a better balance between the various edge cases to keep in mind.

**Comment by FFawzy** [comment]
investigate: do we have information about if there are multiple sections in view and if thats the case what are those sections?

**Comment by gabriele-ct** [comment]
**note** : what written below doesn't address the issue with multiple headings visible when content is at the bottom of the page.

I had a quick look to the code that is handling the highlight and I see that the get the top offset value from `designSystem.dimensions.heights.header`. This is a fixed 48px offset.
However I can see that the actual offset from the top of the viewport to the header is consistently higher, like there's some spacing applied.
Very empirically I tried applying some additional spacing by using the value in `designSystem.dimensions.spacings.m` and that seems to fix the problem of the highlighted selection "out of sync" when clicked. I've tried most of the pages in the smoke test website.

I just wanted to know what you @nkuehn and @emmenko think about it, as you know better the design system

**Comment by nkuehn** [comment]
@gabriele-ct  thanks for the research. Yes, it may well be that we forgot the remove that header-height spacing when we made the top nav header scroll out of view (that was not originally the case). 

But TBD I doubt that change addresses the fundamental problem this issue is about.  The test case in the kit is this link:   
https://commercetools-docs-kit.vercel.app/docs-smoke-test/views/markdown#last-section  (that short "last section" section is especially there to test such cases)

**Comment by gabriele-ct** [comment]
Thank you @nkuehn.  Reading more carefully what reported by @melindafekete in the ticket I believe the fundamental issue she's highlighting is indeed the fact that clicking **any** link in the navigation sidebar always results in the previous link to be highlighted. This is what I tried to address adding the additional spacing to the offset.

My fix doesn't cover the "corner case" of multiple small heading at the bottom of the page, this is still under investigation

**Comment by nkuehn** [comment]
@gabriele-ct thanks for pointing out that circumstance. I really oversaw that, sorry. 

The logic problem here is that the `<section>` is what we have the intersection observer on and the `<section>` itself does not have a margin or padding to the top (because it'd dependent on the first heading level inside it).  

For CSS spec reasons beyond my understanding the top margin of the heading inside the section is not expanding the section area around it.  I guess that would only be the case with padding.  Maybe there is a more elegant way to tell CSS that the section's area should include the margin of the elements inside it?

**Comment by gabriele-ct** [comment]
- Adding the `overflow: auto` property to the `section` element alllow the element to expand including to the heading margin. This seems to help and there's no need to use additional offset.

- Also, the draft PR contains some example code that is able to figure out which and how many sections are in view (this was the original purpose of the investigation).

I think given the two points above we have enough elements to take a better decision on how to address the issue
